### ISSUE TITLE
"sourceDirectory" is not a directory exception

### ISSUE THREAD

**Issue by DexterMagnific** [microsoft/vscode-cmake-tools#1025]
### Cannot configure when toplevel CMakeLists.txt not in wkspc folder

I know this bug seems to be fixed in #533, but I still have it.

Here is our dir structure:

```
projectRoot/
     ide/VSCode/ourProject.code-workspace
     src/CMakeLists.txt
```

the workspace file contains the following:

```
{
	"folders": [
		{
			"path": "../../src"
		},
	],
	"settings": {
		"cmake.sourceDirectory": "{workspaceFolder}/../../src",
		"cmake.buildDirectory": "${workspaceFolder}/../../build"
	}
}
```

### Expected:

1. CMake -> Clean Reconfigure
2. CMake configure is OK

### Apparent Behavior:

Exception:

[rollbar] Unhandled exception: Unhandled Promise rejection: cleanConfigure [cmake-server] Failed to activate protocol version: "sourceDirectory" is not a directory. {}

### Platform and Versions

- **Operating System**: Ubuntu
- **CMake Version**: 3.13.4
- **VSCode Version**: 1.41.1
- **CMake Tools Extension Version**: 1.2.3
- **Compiler/Toolchain**: GCC 8.3

**Comment by bobbrow** [comment]
It looks like you have a typo here:
```
"cmake.sourceDirectory": "{workspaceFolder}/../../src"
```

Can you add a `$` before the `{workspaceFolder}` and see if that fixes it?

**Comment by DexterMagnific** [comment]
Yes this was a typo. I get the same error with `$`. I even removed the variable and left only `../../src` with no success.

What is surprising is that I used to get the same exception but with the directory in the message. Now it is empty.

**Comment by bobbrow** [comment]
Sorry, I gravitated toward that instead of the layout of your code.  Your ${workspaceFolder} is actually `../../src` based on what I see in the code-workspace file.  Can you try just:
```
"cmake.sourceDirectory": "${workspaceFolder}"
```
I created a test with a similar layout and that works for me.  I'm not sure why the error doesn't show the path.  I'll make sure there is something in the log that tells you what the failing folder is.

**Comment by DexterMagnific** [comment]
I tried your suggestion, now I don't have the exception, but a popup saying:
"CMakeLists.txt was not found in the root of the workspace folder, "quickstart a new ...", "edit ...."".

And I don't understand why the workspace folder is src. I opened the workspace file from ide/VSCode so for me this is the workspace folder.

If we sum up things:
- when the sourceDir is "${workspaceFolder}/../../src", I have the exception. I suppose that the extension found the CMakeLists.txt and then something went wrong
- when the sourceDir is "${workspaceFolder}", I have the "CMakeLists.txt" not found in root dir.

Can I help with some logs ?

**Comment by Zingam** [comment]
@DexterMagnific It is always best if you can provide a small sample to reproduce the issue. I have here one that I use if you need as a starting point: https://github.com/microsoft/vscode-cmake-tools/files/4093138/CMakeMultiProject.zip

**Comment by DexterMagnific** [comment]
Thanks for the example. It slightly different from my layout since it contains a CMakeLists.txt in its root.

Anyway, I changed your example to fit my layout:
```
CMakeProject01
   Sources
       CMakeLists.txt
   ide
      settings.code-workspace
```

This works fine. Now I can tell you EXACTLY what is wrong. If you add a LEVEL inside `ide`, (say `ide/vscode/settings.code-workspace`). I have the exception.

**Comment by bobbrow** [comment]
When you open the workspace, are you opening the workspace file, or opening the folder?  I set up my test case exactly as you described.
```
root/
     ide/VSCode/test.code-workspace
     src/CMakeLists.txt
```
When I open the code-workspace file, I see this in the Explorer:
![image](https://user-images.githubusercontent.com/12818240/73474794-ea3da600-4343-11ea-9163-8f9c069e0c76.png)

My settings look like this:
![image](https://user-images.githubusercontent.com/12818240/73474971-57513b80-4344-11ea-8756-e02aa4ef6803.png)

Configure/Build/etc all work for me.  I added logging for the next release that will print out which folder it thinks is the sourceDirectory, so if there is a remaining issue after the release, we can see what is going on.  I hope to publish the release on Monday.
